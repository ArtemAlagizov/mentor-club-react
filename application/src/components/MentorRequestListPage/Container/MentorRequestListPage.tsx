import React, { useEffect } from "react";
import { connect } from "react-redux";
import { useHistory } from "react-router-dom";
import MentorRequestListPageView from "../View/MentorRequestListPageView";
import { fetchMentorRequestsBegin } from "../../../state/MentorRequests/actions";
import { IAppState } from "../../../store";
import {
  selectFetchMentorRequestsError,
  selectIsFetchingMentorRequests,
  selectMentorRequests
} from "../../../state/MentorRequests/selectors";
import { IMentorRequestListPageProps } from "./IMentorRequestListPageProps";

export const MentorRequestListPage: React.FC<IMentorRequestListPageProps> = ({
  fetchMentorRequestsBegin,
  mentorRequests,
  isLoading,
  error
}) => {
  const history = useHistory();

  useEffect(() => {
    fetchMentorRequestsBegin();
  }, [fetchMentorRequestsBegin]);

  const handleRowClick = (key: string) => {
    history.push(`/mentor-requests/${key}/result`);
  };

  return (
    <MentorRequestListPageView
      mentorRequests={mentorRequests}
      isLoading={isLoading}
      error={error}
      handleRowClick={handleRowClick}
    />
  );
};

export const mapStateToProps = (state: IAppState) => ({
  mentorRequests: selectMentorRequests(state),
  isLoading: selectIsFetchingMentorRequests(state),
  error: selectFetchMentorRequestsError(state)
});

export const mapDispatchToProps = {
  fetchMentorRequestsBegin
} as const;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MentorRequestListPage);
