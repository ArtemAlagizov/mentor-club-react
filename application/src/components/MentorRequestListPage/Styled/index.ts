import styled from "styled-components";
import { SpinnerBoundaries } from "./SpinnerBoundaries";

const Container = styled.div`
  padding: 15px;

  .ant-spin-nested-loading {
    height: 200px;
  }
`;

const MentorRequestList = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding-bottom: 20px;
`;

const CardContainer = styled.div`
  padding: 10px;

  .ant-card-bordered {
    border: 0;
  }

  .mentor-request-list-page-card {
    width: 300px;
  }
`;

const UserImage = styled.img`
  margin-top: -3px;
  width: 3rem;
`;

const TopHeaderContainer = styled.div`
  display: flex;
  justify-content: space-between;
  color: white;
`;

const TopHeader = styled.div`
  font-size: 20px;
  color: white;
`;

export const Styled = {
  Container,
  TopHeaderContainer,
  TopHeader,
  SpinnerBoundaries,
  MentorRequestList,
  CardContainer,
  UserImage
};
