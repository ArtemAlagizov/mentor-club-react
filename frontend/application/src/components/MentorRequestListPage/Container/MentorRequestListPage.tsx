import React, { useEffect } from "react";
import { connect } from "react-redux";
import MentorRequestListPageView from "../View/MentorRequestListPageView";
import { fetchMentorRequestsBegin } from "../../../state/MentorRequests/actions";
import { IAppState } from "../../../store";
import {
  selectFetchMentorRequestsError,
  selectIsFetchingMentorRequests,
  selectMentorRequests
} from "../../../state/MentorRequests/selectors";
import { IMentorRequestListPageProps } from "./IMentorRequestListPageProps";

export const MentorRequestListPage: React.FC<IMentorRequestListPageProps> = ({
  fetchMentorRequestsBegin,
  mentorRequests,
  isLoading,
  error
}) => {
  useEffect(() => {
    fetchMentorRequestsBegin();
  }, [fetchMentorRequestsBegin]);

  return (
    <MentorRequestListPageView
      mentorRequests={mentorRequests}
      isLoading={isLoading}
      error={error}
    />
  );
};

export const mapStateToProps = (state: IAppState) => ({
  mentorRequests: selectMentorRequests(state),
  isLoading: selectIsFetchingMentorRequests(state),
  error: selectFetchMentorRequestsError(state)
});

export const mapDispatchToProps = {
  fetchMentorRequestsBegin
} as const;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MentorRequestListPage);
