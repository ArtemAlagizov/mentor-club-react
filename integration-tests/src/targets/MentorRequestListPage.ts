/// <reference types="Cypress" />

export class MentorRequestListPage {
  public static getErrorMessage = () =>
    cy.get('[data-id="fetch-mentor-requests-error"]');

  public static getMentorRequestListTable = () =>
    cy.get('[data-id="mentor-request-list-table"]').children();

  public static getNewMentorRequestButton = () =>
    cy.get('[data-id="new-mentor-request-button"]');

  public static getMentorRequestRow = (rowNumber: number) =>
    MentorRequestListPage.getMentorRequestListTable()
      .get("tr")
      .eq(rowNumber);
}
