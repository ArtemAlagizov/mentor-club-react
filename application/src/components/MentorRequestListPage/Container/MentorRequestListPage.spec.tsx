import React from "react";
import { shallow } from "enzyme";
import { MentorRequestListPage } from "./MentorRequestListPage";

jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: jest.fn()
  })
}));

describe("Given MentorRequestCreationPage", () => {
  it("should have MentorRequestResultPage matching snapshot", () => {
    const component = shallow(
      <MentorRequestListPage
        mentorRequests={[]}
        error={null}
        isLoading={false}
        fetchMentorRequestsBegin={jest.fn()}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
