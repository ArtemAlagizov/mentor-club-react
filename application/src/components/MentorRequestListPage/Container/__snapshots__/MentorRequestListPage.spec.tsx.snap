// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given MentorRequestCreationPage should have MentorRequestResultPage matching snapshot 1`] = `
<MentorRequestListPageView
  error={null}
  isLoading={false}
  mentorRequests={Array []}
/>
`;
